packages/hooks/src/useTrackedEffect/index.ts

```diff
@@ -20,8 +20,8 @@ const diffTwoDeps = (deps1?: DependencyList, deps2?: DependencyList) => {
     : [];
 };
 
-const useTrackedEffect = (effect: Effect, deps?: DependencyList) => {
-  const previousDepsRef = useRef<DependencyList>();
+const useTrackedEffect = <T extends any[]>(effect: Effect<T>, deps?: [...T]) => {
+  const previousDepsRef = useRef<T>();
```

Comment:
how about replacing `T extends any[]` with `T extends unknown[]`?

Reply:
`unknown` is better and code is updated.